driver/docker/version.go

```diff
@@ -0,0 +1,106 @@
+package docker
+
+import (
+	"github.com/Masterminds/semver/v3"
+)
+
+type mobyBuildkitVersion struct {
+	MobyVersionConstraint string
+	BuildkitVersion       string
+}
+
+// https://github.com/Masterminds/semver#checking-version-constraints
+var mobyBuildkitVersions = []mobyBuildkitVersion{
+	{
+		MobyVersionConstraint: "18.06.0-0 - 18.09.0-0",
+		BuildkitVersion:       "v0.3.0",
+	},
+	{
+		MobyVersionConstraint: "18.09.1-0 - 18.09.9-0",
+		BuildkitVersion:       "v0.4.0",
+	},
+	{
+		MobyVersionConstraint: "19.03.0-0 - 19.03.1-0",
+		BuildkitVersion:       "v0.6.0",
+	},
+	{
+		MobyVersionConstraint: "19.03.2-0 - 19.03.6-0",
+		BuildkitVersion:       "v0.6.2",
+	},
+	{
+		MobyVersionConstraint: "~19.03-0",
+		BuildkitVersion:       "v0.6.4",
+	},
+	{
+		MobyVersionConstraint: "20.10.0-0 - 20.10.2-0",
+		BuildkitVersion:       "v0.8.1",
+	},
+	{
+		MobyVersionConstraint: ">= 20.10.3-0, < 20.10.4-0",
+		BuildkitVersion:       "v0.8.2+68bb095",
+	},
+	{
+		MobyVersionConstraint: "20.10.4-0 - 20.10.6",
+		BuildkitVersion:       "v0.8.2",
+	},
+	{
+		MobyVersionConstraint: "20.10.7-0 - 20.10.10-0",
+		BuildkitVersion:       "v0.8.3+244e8cde",
+	},
+	{
+		MobyVersionConstraint: "20.10.11-0 - 20.10.18-0",
+		BuildkitVersion:       "v0.8.3+bc07b2b8",
+	},
+	{
+		MobyVersionConstraint: ">= 20.10.19-0, < 20.10.20-0",
+		BuildkitVersion:       "v0.8.3+3a1eeca5",
+	},
+	{
+		MobyVersionConstraint: ">= 20.10.20-0, < 20.10.21-0",
+		BuildkitVersion:       "v0.8.3+c0149372",
+	},
+	{
+		MobyVersionConstraint: ">= 20.10.21-0, <= 20.10.23",
+		BuildkitVersion:       "v0.8.3+eeb7b65",
+	},
+	{
+		MobyVersionConstraint: "~20.10-0",
+		BuildkitVersion:       "v0.8+unknown",
+	},
+	{
+		MobyVersionConstraint: "~22.06-0",
+		BuildkitVersion:       "v0.10.3",
+	},
+	{
+		MobyVersionConstraint: ">= 23.0.0-0, < 23.0.1-0",
+		BuildkitVersion:       "v0.10.6",
+	},
+	{
+		MobyVersionConstraint: "23.0.1",
+		BuildkitVersion:       "v0.10.7+4f0ee09",
+	},
+	{
+		MobyVersionConstraint: "~23-0",
+		BuildkitVersion:       "v0.10+unknown",
+	},
+}
```

Comment:
Switch to [version constraint](https://github.com/Masterminds/semver#checking-version-constraints) with hyphen and tilde ranges comparisons. Let me know if that SGTY.